src/jdk.jlink/share/classes/jdk/tools/jlink/internal/runtimelink/RuntimeImageLinkException.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2024, Red Hat, Inc.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package jdk.tools.jlink.internal.runtimelink;
+
+/**
+ * Exception thrown for links without packaged modules. I.e. run-image link.
+ *
+ */
+public class RuntimeImageLinkException extends IllegalStateException {
+
+    private static final long serialVersionUID = -1848914673073119403L;
+
+    private final IllegalArgumentException iae;
+
+    public RuntimeImageLinkException(IllegalArgumentException cause) {
```

Comment:
Can you explain why `RuntimeImageLinkException` constructor takes IAE.   The places throwing RILE has to create IAE.  I can't see how IAE is necessary as other places would catch RILE and throw IAE or IOException instead.  

Can it simply take an exception message parameter?   And possibly change `JlinkTask::run` in handling RILE like IAE?

```
        } catch (IllegalArgumentException | ResolutionException | RuntimeImageLinkException e) {
            log.println(taskHelper.getMessage("error.prefix") + " " + e.getMessage());
            if (DEBUG) {
                e.printStackTrace(log);
            }
            return EXIT_ERROR;
 ```

Reply:
This was left-over from previous code. It's now just taking a string.